services/device-registry-base/src/main/java/org/eclipse/hono/service/management/device/DelegatingDeviceManagementHttpEndpoint.java

```diff
@@ -135,6 +143,39 @@ private void doGetDevice(final RoutingContext ctx) {
             .onComplete(s -> span.finish());
     }
 
+    private void doSearchDevices(final RoutingContext ctx) {
+        final Span span = TracingHelper.buildServerChildSpan(
+                tracer,
+                TracingHandler.serverSpanContext(ctx),
+                SPAN_NAME_SEARCH_DEVICES,
+                getClass().getSimpleName()).start();
+
+        final Future<String> tenantId = getRequestParameter(ctx, PARAM_TENANT_ID,
+                getPredicate(config.getTenantIdPattern(), false));
+        final Optional<Integer> pageSize = Optional
+                .ofNullable(ctx.request().getParam(RegistryManagementConstants.FIELD_PAGE_SIZE))
+                .map(Integer::parseInt);
```

Comment:
what if it's a float?

Reply:
In the spec, the required _data type_ for the `pageSize` is mentioned as _integer_. If it is _float_, then `NumberFormatException` will be thrown. I didn't get the intention of the comment. Is it to handle this exception using `ClientErrorExecption`?

Reply:
When I run a curl command on my machine against your implementation:
```
curl http://registry/v1/devices/my-tenant?pageSize=23.5
```
what will happen?

Reply:
It will return _500 Internal Server Error_. I will handle those exceptions so that  _HTTP_BAD_REQUEST_ is returned.